hspec-discover/test/RunSpec.hs

```diff
@@ -83,6 +83,12 @@ spec = do
 
     it "returns Nothing for invalid spec name" $ do
       fileToSpec "" "foo" `shouldBe` Nothing
+      fileToSpec "" "fooSpec.hs" `shouldBe` Nothing
+      fileToSpec "" "flycheck_Spec.hs" `shouldBe` Nothing
+      fileToSpec "" ("Valid" </> "invalid"  </>"MiddleNamesSpec.hs")
+          `shouldBe` Nothing
+      fileToSpec "" "Weird.Spec.hs" `shouldBe` Nothing
+      fileToSpec "/" "Spec.hs" `shouldBe` Nothing
```

Comment:
Let's stick with one expectation per test.  I would be happy with two cases here:

```
context "when spec does not have a valid module name" $ do
  it "returns Nothing" $ do
    fileToSpec "" "flycheck_Spec.hs" `shouldBe` Nothing

context "when any component of a hierarchical module name is not valid" $ do
  it "returns Nothing" $ do
    fileToSpec "" ("Valid" </> "invalid" </>"MiddleNamesSpec.hs") `shouldBe` Nothing
```

If we want to tests for a wide variety of invalid module names, then that should be tests for `isValidModuleName`.  But I would not do so, as we copy & pasted `isValidModuleName` anyway, and I mostly write tests to drive the implementation.